tests/integration/src/test/java/org/mule/test/integration/domain/http/HttpSharePortTestCase.java

```diff
@@ -46,25 +52,26 @@ protected String getDomainConfig() {
 
   @Test
   public void bothServicesBindCorrectly() throws Exception {
-    InternalMessage helloWorldServiceResponse = getMuleContextForApp(HELLO_WORLD_SERVICE_APP).getClient()
-        .send(format("%s://localhost:%d/service/helloWorld", endpointScheme.getValue(), dynamicPort.getNumber()),
-              InternalMessage.builder().payload("test-data").build(), getOptionsBuilder().build())
-        .getRight();
-    assertThat(getPayloadAsString(helloWorldServiceResponse, getMuleContextForApp(HELLO_WORLD_SERVICE_APP)), is("hello world"));
+    HttpRequest httpRequest = HttpRequest.builder()
+        .setUri(format("%s://localhost:%d/service/helloWorld", endpointScheme.getValue(), dynamicPort.getNumber())).build();
+    HttpResponse httpResponse = httpClient.send(httpRequest, RECEIVE_TIMEOUT, false, null);
+
+    String payload = IOUtils.toString(((InputStreamHttpEntity) httpResponse.getEntity()).getInputStream(), UTF_8);
+    assertThat(payload, is("hello world"));
 
-    InternalMessage helloMuleServiceResponse = getMuleContextForApp(HELLO_MULE_SERVICE_APP).getClient()
-        .send(format("%s://localhost:%d/service/helloMule", endpointScheme.getValue(), dynamicPort.getNumber()),
-              InternalMessage.builder().payload("test-data").build(), getOptionsBuilder().build())
-        .getRight();
-    assertThat(getPayloadAsString(helloMuleServiceResponse, getMuleContextForApp(HELLO_MULE_SERVICE_APP)), is("hello mule"));
+    httpRequest = HttpRequest.builder()
+        .setUri(format("%s://localhost:%d/service/helloMule", endpointScheme.getValue(), dynamicPort.getNumber())).build();
+    httpResponse = httpClient.send(httpRequest, RECEIVE_TIMEOUT, false, null);
+
+    payload = IOUtils.toString(((InputStreamHttpEntity) httpResponse.getEntity()).getInputStream(), UTF_8);
+    assertThat(payload, is("hello mule"));
   }
 
   protected SystemProperty getEndpointSchemeSystemProperty() {
     return new SystemProperty("scheme", "http");
   }
 
-  protected HttpRequestOptionsBuilder getOptionsBuilder() {
-    return newOptions();
+  protected TlsContextFactory getTlsContextFactory() {
```

Comment:
is it needed for this to return null or can this just be removed? (and the call on the builder too)